app/models/manageiq/providers/infra_manager.rb

```diff
@@ -7,6 +7,18 @@ class InfraManager < BaseManager
 
     include AvailabilityMixin
 
+    has_many :host_hardwares,         :through => :hosts, :source => :hardware
+    has_many :host_operating_systems, :through => :hosts, :source => :operating_system
+    has_many :host_storages,          :through => :hosts
+    has_many :host_switches,          :through => :hosts
+    has_many :host_networks,          :through => :hosts, :source => :networks
+    has_many :snapshots,              :through => :vms_and_templates
+    has_many :operating_systems,      :through => :vms_and_templates
+    has_many :switches,               :through => :hardwares
+    has_many :lans,                   :through => :hardwares
```

Comment:
please see `:host_switches` in line 13 it goes through `:hosts` and `:switches`/`:lans` are used by the vms. Is it OK with you?